core/server/src/main/java/alluxio/worker/netty/BlockDataServerHandler.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.netty;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.StorageTierAssoc;
+import alluxio.WorkerStorageTierAssoc;
+import alluxio.exception.BlockDoesNotExistException;
+import alluxio.exception.InvalidWorkerStateException;
+import alluxio.network.protocol.RPCBlockReadRequest;
+import alluxio.network.protocol.RPCBlockReadResponse;
+import alluxio.network.protocol.RPCBlockWriteRequest;
+import alluxio.network.protocol.RPCBlockWriteResponse;
+import alluxio.network.protocol.RPCResponse;
+import alluxio.network.protocol.databuffer.DataBuffer;
+import alluxio.network.protocol.databuffer.DataByteBuffer;
+import alluxio.network.protocol.databuffer.DataFileChannel;
+import alluxio.worker.block.BlockWorker;
+import alluxio.worker.block.io.BlockReader;
+import alluxio.worker.block.io.BlockWriter;
+
+import com.google.common.base.Preconditions;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * This class handles {@link RPCBlockReadRequest}s and {@link RPCBlockWriteRequest}s.
+ */
+@NotThreadSafe
+public final class BlockDataServerHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  /** The Block Worker which handles blocks stored in the Alluxio storage of the worker. */
+  private final BlockWorker mWorker;
+  /** The transfer type used by the data server. */
+  private final FileTransferType mTransferType;
+  /**  An object storing the mapping of tier aliases to ordinals. */
+  private final StorageTierAssoc mStorageTierAssoc;
+
+  protected BlockDataServerHandler(BlockWorker worker, Configuration configuration) {
+    mWorker = worker;
```

Comment:
Only in the constructor, and when no method is called on the input. This will catch it early and throw the call stack of the context.